service/history/historyEngine.go

```diff
@@ -3009,6 +3009,21 @@ func (e *historyEngineImpl) ReapplyEvents(
 				}, nil
 			}
 
+			// Check if max signal count reaches
+			// Currently, we only re apply signal events
+			executionInfo := mutableState.GetExecutionInfo()
+			maxAllowedSignals := e.config.MaximumSignalsPerExecution(domainEntry.GetInfo().Name)
+			if maxAllowedSignals > 0 && (int(executionInfo.SignalCount)+len(toReapplyEvents)) >= maxAllowedSignals {
+				e.logger.Info("Execution limit reached for maximum signals during signal re-application",
```

Comment:
This seems like a quite critical error, as it's non-retryable and the passive side doesn't even know the reapply has failed. Shall we use `Error` level log and probably emit some metrics?

Reply:
signal reapplication should ignore this limit.
or at least return error

Reply:
Remove this check in this PR. If it returns an error, the caller side has. to handle this error or it will retry forever. I will include it in a different PR.